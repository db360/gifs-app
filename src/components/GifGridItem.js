import propTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    // console.log(id, title, url);

    return (
        <div className="card animate__fadeInLeftBig animate__fadeIn">
            <img src={url} alt={title} />
            <p> { title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}