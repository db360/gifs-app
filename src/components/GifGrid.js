// import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

        return (
            <>
                <h3 className="animate__pulse"> { category } </h3>

                { loading && <p className="animate__swing">Loading</p> }

                <div className="card-grid">
                
                    {
                        images.map( img => (
                            <GifGridItem
                                key={ img.id } 
                                { ...img }
                            />
                        ))
                    }
                
                </div>
            </> 
    )
}
