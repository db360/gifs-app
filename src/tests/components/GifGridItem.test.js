import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas a <GifGridItem >', () => {
    
    const title = 'Un TÃ­tulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url}/>)

    test('Se debe de mostrar el componente correctamente ', () => {
        
        expect( wrapper).toMatchSnapshot();

    });

    test('Debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);

    });

    test('Debe tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        // console.log(img.props('src'));
        expect(img.prop('src') ).toBe(url);
        expect(img.prop('alt') ).toBe(title);

    });

    test('Debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        // console.log(div.props('className'))
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );

    });
    
    
})
