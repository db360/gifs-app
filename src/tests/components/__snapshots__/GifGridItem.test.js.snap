// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGridItem > Se debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__fadeInLeftBig animate__fadeIn"
>
  <img
    alt="Un Título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un Título
  </p>
</div>
`;
